hudi-common/src/main/java/org/apache/hudi/common/config/DefaultHoodieConfig.java

```diff
@@ -26,6 +26,11 @@
  */
 public class DefaultHoodieConfig implements Serializable {
 
+  public static final String QUERY_TYPE_OPT_KEY = "hoodie.datasource.query.type";
```

Comment:
`HiveSyncTool` need to refer these config. However, they are located in the `hudi-spark` module which cannot be used in the `hudi-sync` module. So I push down these config to the `hudi-common`

Reply:
I agree with @vinothchandar that if possible the common module should be unaware of the query types.

Instead of storing the `hoodie.datasource.query.type` in the table properties can we instead store the table name in the table properties along with its suffix `ro` or `rt` and make the decision based on that ? With this approach, I see other problems as well like if someone goes and by mistake changes the `hoodie.datasource.query.type` property for the `ro` table it can start behaving like real time table. To avoid such things, I think making this decision based on table name itself (as we have been doing so far) seems better.

Reply:
yeah, I don't see any usages for this config outside of HiveSyncTool. So, better to keep it local. And Initially I thought on similar lines as @umehrot2 that we can leverage table name to detect. but later realized that, this particular config is used while syncing schema and the caller hard codes the type(there is only one caller to this method and is local to HoodieSyncTool). So, may not be required to leverage table name.